### ISSUE TITLE
Client looses connection after some time?

### ISSUE THREAD

**Issue by johnknoop** [Azure/azure-service-bus#52]
Hi

I've basically just followed the examples and created a console app that runs inside a docker container on one of my azure vm's:

```            
queueClient.RegisterSessionHandler(async (session, message, cancel) =>
            {
                var greeting = message.Body.As<Greeting>();
                Console.WriteLine($"Saving greeting with message {greeting.Message}");
                await SaveGreeting(greeting);
                await session.CompleteAsync(message.SystemProperties.LockToken);
            });

            Console.ReadKey(true);
```

`SaveGreeting` will just put store the message in a database.

Now what I do is, every couple of days, I just publish a message on the queue, and check to see that it's been persisted to the database. During the first couple of days after starting the program that listens for messages, everything works. But after 5-6 days, it seems to loose connection or something, because the messages stay on the queue. Then, as soon as I restart the program, the messages gets consumed and stored to the db. It always seems to be 5-6 days of inactivity that triggers the "connection loss" or whatever it is.

Could it be that I somehow misunderstood how to use this library, or might it be a bug in there?

Thanks

**Comment by binzywu** [comment]
@johnknoop Is this for the new .netstd library?

**Comment by johnknoop** [comment]
No this was before the RC1 was released. I haven't been able to deploy a new version since that one came out, because of an issue with VSTS and dotnet core on linux build servers. I've been told by the folk over at the VSTS repo that they have a fix that will be in place within two weeks.

**Comment by gastart** [comment]
I am facing more or less the same issue. Issue occurred with 3.1.0 and 3.1.1 using .net standard.  Never used service bus before so I cannot tell how older libraries are behaving.

My setup is that I have a ServicebusConnection and 5 message receivers using it.  I have a polling mechanism with a timeout of 10 seconds on each running in parallel.

After some days running "fine", I observed that the queue that has the highest activity (more or less 300 messages every 5 minutes) was not being consumed anymore. My client code was reporting that no messages being received.  Other queues with far less activity seemed to be working ok.  I restarted my service and the queued started to be consumed.

@binzywu are there issues with .net standard library?  Is there some form of tracing that I can set so as to better troubleshoot the issue?
